Source/gmenu.cpp

```diff
@@ -93,37 +93,27 @@ BOOL gmenu_exception()
 	return sgpCurrentMenu != 0;
 }
 
-void gmenu_call_proc(TMenuItem *pItem, void(*gmFunc)(TMenuItem *))
+void gmenu_call_proc(TMenuItem *pItem, void (*gmFunc)(TMenuItem *))
 {
-	TMenuItem *v2;         // eax
-	int v3;                // ecx
-	void(* *v4)(BOOL); // edx
+	void (**v4)(BOOL);
 
 	PauseMode = 0;
-	mouseNavigation = 0;
-	v2 = pItem;
-	dword_63447C = gmFunc;
+	mouseNavigation = FALSE;
 	sgpCurrentMenu = pItem;
+	dword_63447C = gmFunc;
 	if (gmFunc) {
-		gmFunc(sgpCurrentMenu);
-		v2 = sgpCurrentMenu;
+		dword_63447C(sgpCurrentMenu);
+		pItem = sgpCurrentMenu;
 	}
-	v3 = 0;
 	sgCurrentMenuIdx = 0;
-	if (v2) {
-		v4 = &v2->fnMenu;
-		while (*v4) {
-			++v3;
-			v4 += 3;
-			sgCurrentMenuIdx = v3;
+	if (sgpCurrentMenu) {
+		for (v4 = &sgpCurrentMenu->fnMenu; *v4; v4 += 3) {

```

Comment:
is it possible that this was?
```suggestion
		for (v3 = 0; sgpCurrentMenu[v3].fnMenu; v3++) {
			sgCurrentMenuIdx = v3;
```

Reply:
yeah, still bin exact, nice :D